projects/packages/forms/src/dashboard/inbox/list.js

```diff
@@ -13,6 +14,7 @@ const COLUMNS = [
 	{
 		key: 'date',
 		label: __( 'Date', 'jetpack-forms' ),
+		getValue: item => dateI18n( 'M j, Y', item.date ),
```

Comment:
Should the date format be localized, like we do here?
https://github.com/Automattic/jetpack/blob/a05d65b2ed858afe5710e43d17ee07f6b6e79947/projects/packages/forms/src/dashboard/inbox/response.js#L66

Or maybe that wouldn't look that good in the UI, I'm not sure?